oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -12,43 +31,131 @@ fn escape_globing_pattern(string: &str) -> String {
 }
 
 impl Regex {
-    pub fn from_pattern(pattern: &str) -> Result<Self, fnmatch_regex::error::Error> {
-        let escaped_str = escape_globing_pattern(pattern);
-        Ok(Regex(fnmatch_regex::glob_to_regex(&escaped_str)?))
+    /// Returns a regex which is built from a file that contains shell like patterns
+    pub fn from_file(file_path: &Path) -> RegexResult {
+        let file_content =
+            fs::read_to_string(file_path).map_err(|err| RegexError::PatternFileIOError { err })?;
+
+        Ok(Self(
+            file_content
+                .lines()
+                .into_iter()
+                .filter(|l| l.trim() != "\n" && !l.trim().starts_with('#'))
+                .map(|l| {
+                    fnmatch_regex::glob_to_regex(&escape_globing_pattern(l.trim()))
+                        .map_err(|err| RegexError::GlobPatternError { err })
+                })
+                .collect::<Result<Vec<regex::Regex>, RegexError>>()?,
+        ))
     }
 
-    pub fn from_regex_str(regex_str: &str) -> Result<Self, regex::Error> {
-        Ok(Regex(regex::Regex::new(regex_str)?))
+    /// Returns a regex which is an union of all regexes from `raw_regexes` slice parameter
+    pub fn from_raw_regexes(raw_regexes: &[&str]) -> RegexResult {
+        Ok(Self(
+            raw_regexes
+                .iter()
+                .map(|l| regex::Regex::new(l).map_err(|err| RegexError::ParseError { err }))
+                .collect::<Result<Vec<regex::Regex>, RegexError>>()?,
+        ))
     }
 
-    pub fn is_match(&self, string: &str) -> bool {
-        self.0.is_match(string)
+    /// Returns a regex from a glob pattern
+    pub fn from_glob_pattern(pattern: &str) -> RegexResult {
+        let escaped_str = escape_globing_pattern(pattern);
+        Ok(Regex(vec![fnmatch_regex::glob_to_regex(&escaped_str)
+            .map_err(|err| RegexError::GlobPatternError { err })?]))
+    }
+
+    pub fn from_regex_str(regex_str: &str) -> RegexResult {
+        Self::from_raw_regexes(&[regex_str])
     }
-}
 
-impl AsRef<str> for Regex {
-    fn as_ref(&self) -> &str {
-        self.0.as_str()
+    pub fn is_match(&self, string: &str) -> bool {
+        self.0.iter().any(|r| r.is_match(string))
     }
 }
 
 impl PartialEq for Regex {
     fn eq(&self, other: &Self) -> bool {
-        self.as_ref() == other.as_ref()
+        if self.0.len() == other.0.len() {
+            let set: HashSet<&str> = HashSet::from_iter(self.0.iter().map(|r| r.as_str()));
+
+            other.0.iter().all(|r| set.contains(r.as_str()))
+        } else {
+            false
+        }
```

Comment:
uh ? Why do you use `Vec` then ?

Reply:
Sorry, I don't understand. What did you mean?

If your questions is why did I chose a `Vec`, I took it because `Regex` isn't hashable. This comparison is only used in a few python tests.

Reply:
Can you add a comment then ?
Because another solution is to implement hash with his str repr

Reply:
I've added a comment